Lib/test/test_module.py

```diff
@@ -346,6 +346,25 @@ def test_repeated_attribute_pops(self):
 
     # frozen and namespace module reprs are tested in importlib.
 
+    def test_subclass_with_slots(self):
+        # In 3.11alpha this crashed, as the slots weren't NULLed.
+
+        class ModuleWithSlots(ModuleType):
+            __slots__ = ("a", "b")
+
+            def __init__(self, name):
+                super().__init__(name)
```

Comment:
What's the purpose of this method? Maybe assign a and/or b attribute if you want to keep it? Or remove it?

Reply:
The important bit is having ``__slots__`` in the class definition, I just checked and leaving out the ``__init__`` method still reproduces the crash that this PR fixes.

Reply:
Well, I approved the PR. @markshannon can decide to keep or remove the method, it doesn't really matter ;-)